tiles/src/main/java/com/protomaps/basemap/layers/Places.java

```diff
@@ -25,6 +29,17 @@ public String name() {
 
   private final AtomicInteger placeNumber = new AtomicInteger(0);
 
+  // To have consistent min_zoom and other attr between Natural Earth (low zooms) and OpenStreetMap (high zooms)
+  // we need to store the NE places with a spatial indexes for later joins with OSM
+  private PointIndex<NaturalEarthPlace> ne_populated_places = PointIndex.create();
+
+  // Data structure for any Natural Earth place (eg for data joins between NE and OSM)
+  private record NaturalEarthPlace(String name, String wikidataId, float minZoom, int populationRank) {}
+
+  // Search window size for NE <> OSM data joins
+  private static final double LOCALITY_JOIN_DISTANCE = GeoUtils.metersToPixelAtEquator(0, 50_000) / 256d;
```

Comment:
from what I read below, this is only necessary in 2% of the cases that don't have a Wikidata ID?